### PR TITLE
Feature/chats

### PR DESCRIPTION
**Pull request by Vestanra**

### CODE REVIEW

File: src/services/socket.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { NotificationEvents } from "../interfaces-submodule/enums/notification/notification-events.enum";
+import { BaseRoutes } from "../interfaces-submodule/enums/routes/base-routes.enum";
+import { MessagesRoutesEnum } from "../interfaces-submodule/enums/routes/messages-routes.enum";
+import { ISubscriptionToChatMessagesDTO } from "../interfaces-submodule/interfaces/dto/message/isubscription-to-chat-messages.dto";
+
+export const socketUrl = `${process.env.REACT_APP_BASE_URL}${BaseRoutes.V1}/${MessagesRoutesEnum.BasePrefix}`;
+
```
Comment (AndriiKasian) [inline src/services/socket.ts]: - [x] use import { Manager } from 'socket.io-client'; to create socketManager

File: src/components/Chats/ChaItem.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        }        
+    }, [chatsData, isFetching]);
+
+    useEffect(() => {
+        setIsFetching(false);
+        setMyMessage('');
+    },[chatsData])
+
+    useEffect(() => {
+        const socket: Socket = io(socketUrl, {
```
Comment (AndriiKasian) [inline src/components/Chats/ChaItem.tsx]: - [x] this part of socket should be in the socket manager

File: src/components/Chats/ChaItem.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        });
+
+        const requestBody = {
+            chatId: parseInt(id || '0'),
+            accessToken
+        };
+
+        socket.on('connect', () => handleConnect(socket, requestBody));
+        socket.on('disconnect', () => handleDisconnect(socket, requestBody));
+        socket.on(NotificationEvents.ChatResponse, () => {
```
Comment (AndriiKasian) [inline src/components/Chats/ChaItem.tsx]: - [x] on NotificationEvents.ChatResponse you just need to add response message to the chat list, but not refetch

File: src/components/Chats/ChaItem.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            transports: ["websocket"],
+            timeout: 10000,
+        });
+
+        const requestBody = {
+            chatId: parseInt(id || '0'),
+            accessToken
+        };
+
+        socket.on('connect', () => handleConnect(socket, requestBody));
```
Comment (AndriiKasian) [inline src/components/Chats/ChaItem.tsx]: - [x] move all logic related to the socket functionality to the custom hook

**Review by AndriiKasian — CHANGES_REQUESTED** [review]

**Comment by Vestanra** [comment]
add socket manager, update chat message logic, add custom hooks for socket and data fetching

**Review by AndriiKasian — APPROVED** [review]

**Pull request merged**